src/analyses/loopTermination.ml

```diff
@@ -0,0 +1,107 @@
+(** Termination analysis for loops and [goto] statements ([termination]). *)
+
+open Analyses
+open GoblintCil
+open TerminationPreprocessing
+
+(** Stores the result of the query whether the program is single threaded or not
+    since finalize does not have access to ctx. *)
+let single_thread : bool ref = ref false
+
+(** Contains all loop counter variables (varinfo) and maps them to their corresponding loop statement. *)
+let loop_counters : stmt VarToStmt.t ref = ref VarToStmt.empty
+
+(** Checks whether a variable can be bounded. *)
+let check_bounded ctx varinfo =
+  let open IntDomain.IntDomTuple in
+  let exp = Lval (Var varinfo, NoOffset) in
+  match ctx.ask (EvalInt exp) with
+  | `Top -> false
+  | `Lifted v -> not (is_top_of (ikind v) v)
```

Comment:
This is now problematic as far as I can see. We execute `set_string "sem.int.signed_overflow" "assume_none";` in `maingoblint.ml` and still rely on an overflow yielding top here. This may currently work due to some peculiarities of the implementation, but there is no reason to believe that it always will.

Reply:
The type of the counter variables introduced is now unsigned, and thus they are not affected by the setting of `sem.int.signed_overflow assume_none`. So it should no longer be a problem.